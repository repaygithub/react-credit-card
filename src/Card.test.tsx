import * as React from 'react'
import { cleanup, render } from 'react-testing-library'
import ReactCreditCard from './Card'

afterEach(cleanup)

describe('React Credit Card', () => {
  test('should match snapshot', () => {
    const creditCard = render(<ReactCreditCard number="" name="" placeholderName="FULL NAME" />)
    expect(creditCard.asFragment()).toMatchSnapshot()
  })

  test('should match amex snapshot', () => {
    const creditCard = render(
      <ReactCreditCard number="340000000000009" name="" placeholderName="FULL NAME" />
    )
    expect(creditCard.asFragment()).toMatchSnapshot()
  })

  test('should render number', () => {
    const creditCard = render(
      <ReactCreditCard number="5105105105105100" name="" placeholderName="FULL NAME" />
    )
    expect(creditCard.getByText('5105 1051 0510 5100')).not.toBeNull()
  })

  test('should show back when focused = cvc', () => {
    let creditCard = render(
      <ReactCreditCard number="" focused="cvc" name="" placeholderName="FULL NAME" />
    )
    expect(creditCard.container.firstChild.firstChild).toHaveClass('ReactCreditCard--flipped')
    creditCard = render(
      <ReactCreditCard number="" focused="number" name="" placeholderName="FULL NAME" />
    )
    expect(creditCard.container.firstChild.firstChild).not.toHaveClass('ReactCreditCard--flipped')
  })

  test('should update name', () => {
    let creditCard = render(<ReactCreditCard number="" name="" placeholderName="FULL NAME" />)
    expect(creditCard.getByText('FULL NAME')).not.toBeNull()
    creditCard = render(<ReactCreditCard number="" name="Eddard Stark" placeholderName="FULL NAME" />)
    expect(creditCard.getByText('Eddard Stark')).not.toBeNull()
  })
})
