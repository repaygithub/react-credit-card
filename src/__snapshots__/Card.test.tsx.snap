// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`React Credit Card should match amex snapshot 1`] = `
<DocumentFragment>
  <div
    class="ReactCreditCard__container"
  >
    <div
      class="ReactCreditCard "
    >
      <div
        class="ReactCreditCard__front"
      >
        <div
          class="ReactCreditCard__lower"
        >
          <div
            class="ReactCreditCard__shiny"
          />
          <img
            class="ReactCreditCard__logo"
            src=""
          />
          <div
            class="ReactCreditCard__number ReactCreditCard__display"
          >
            3400 000000 00009
          </div>
          <div
            class="ReactCreditCard__name ReactCreditCard__display"
          >
            FULL NAME
          </div>
        </div>
      </div>
      <div
        class="ReactCreditCard__back"
      >
        <div
          class="ReactCreditCard__bar"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`React Credit Card should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="ReactCreditCard__container"
  >
    <div
      class="ReactCreditCard "
    >
      <div
        class="ReactCreditCard__front"
      >
        <div
          class="ReactCreditCard__lower"
        >
          <div
            class="ReactCreditCard__shiny"
          />
          <img
            class="ReactCreditCard__logo"
            src=""
          />
          <div
            class="ReactCreditCard__number ReactCreditCard__display"
          >
            •••• •••• •••• •••• 
          </div>
          <div
            class="ReactCreditCard__name ReactCreditCard__display"
          >
            FULL NAME
          </div>
        </div>
      </div>
      <div
        class="ReactCreditCard__back"
      >
        <div
          class="ReactCreditCard__bar"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
